services/obs/obs.service.js

```diff
@@ -0,0 +1,94 @@
+import Joi from 'joi'
+import { BaseXmlService } from '../index.js'
+import { optionalUrl } from '../validators.js'
+import { isBuildStatus, renderBuildStatusBadge } from './obs-build-status.js'
+
+const schema = Joi.object({
+  status: Joi.object({
+    '@_code': isBuildStatus,
+  }).required(),
+}).required()
+
+export default class ObsService extends BaseXmlService {
+  static category = 'build'
+  static route = {
+    base: 'obs',
+    pattern: ':project/:packageName/:repository/:arch',
+    queryParamSchema: Joi.object({
+      instance: optionalUrl,
+    }).required(),
+  }
+
+  static auth = {
+    userKey: 'obs_userName',
+    passKey: 'obs_userPass',
+    authorizedOrigins: [],
+    isRequired: true,
+  }
+
+  static examples = [
+    {
+      title: 'openSUSE OBS package build status',
+      namedParams: {
+        project: 'openSUSE:Tools',
+        packageName: 'osc',
+        repository: 'Debian_11',
+        arch: 'x86_64',
+      },
+      staticPreview: this.render({
+        repository: 'Debian_11',
+        status: 'succeeded',
+      }),
+      keywords: ['open build service'],
+    },
```

Comment:
We should also add another example that includes the `queryParams` key so that there's an example listing on the main badge that allows users to leverage the custom instance url in the badge builder window. Note the actual value doesn't matter for `instance` and it's fine to use the reference server, just as long as there's an example with something like:

```js
queryParams: { instance: 'https://api.opensuse.org' },
```